ogre/src/OgreMaterial.cc

```diff
@@ -351,6 +351,22 @@ void OgreMaterial::UpdateShaderParams(ConstShaderParamsPtr _params,
       name_param.second.Value(&value);
       _ogreParams->setNamedConstant(name_param.first, value);
     }
+    else if (ShaderParam::PARAM_FLOAT_BUFFER == name_param.second.Type())
+    {
+      std::shared_ptr<void> wordBuffer;
+      name_param.second.WordBuffer(&wordBuffer);
+      uint32_t count = name_param.second.Count();
+      _ogreParams->setNamedConstant(name_param.first,
+          reinterpret_cast<float*>(wordBuffer.get()), count, 1);
```

Comment:
can you add a note about the last param? the [API doc](https://www.ogre3d.org/docs/api/1.9/class_ogre_1_1_gpu_program_parameters.html#a655b9ba0879631d9d4bae5d10cc2101e) says only glsl supports a number other than 4 so we should keep that in mind in case if we ever decide to port to metal.